test/functionalTest/cases/2507_metadata_flags_in_notifications/create_and_update_csubs_with_metadata_flags_field.test

```diff
@@ -0,0 +1,326 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Create and update csubs with metatata flags field
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+
+--SHELL--
+
+#
+# 01. Create subscription with metadataFlags: foo, see error
+# 02. Create subscription with metadataFlags: true
+# 03. Get subscription and see metadataFlags: true
+# 04. Update subscription with metadataFlags: bar, see error
+# 05. Update subscription without touching metadataFlags
+# 06. Get subscription and see metadataFlags: true
+# 07. Update subscription with metadataFlags: false
+# 08. Get subscription and see no metadataFlags
+#
+
+echo "01. Create subscription with metadataFlags: foo, see error"
+echo "=========================================================="
+payload='
+{
+    "subject": {
+        "entities": [
+            {
+                "id": "E1",
+                "type": "T"
+            }
+        ],
+        "condition": {
+            "attrs": [ ]
+         }
+    },
+    "notification": {
+        "http": {"url": "http://localhost:1234/notify"},
+        "metadataFlags": "foo",
+        "attrs": [ ]
+    },
+    "expires": "2050-04-05T14:00:00.00Z"
+}
+'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "02. Create subscription with metadataFlags: true"
+echo "================================================"
+payload='
+{
+    "subject": {
+        "entities": [
+            {
+                "id": "E1",
+                "type": "T"
+            }
+        ],
+        "condition": {
+            "attrs": [ ]
+         }
+    },
+    "notification": {
+        "http": {"url": "http://localhost:1234/notify"},
+        "metadataFlags": true,
```

Comment:
"details" is too generic...

The name should be something that give the following idea "setting this to true you would get metadata marks in the notifications associated to this subscription". Which short and compact name you suggest for that?